src/main/java/de/tum/in/www1/artemis/service/competency/LearningObjectImportService.java

```diff
@@ -0,0 +1,436 @@
+     *
+     * @param sourceCourseCompetencies The source course competencies to import from.
+     * @param idToImportedCompetency   A map from the source competency IDs to the imported competencies.
+     * @param courseToImportInto       The course to import the learning objects into.
+     * @param importOptions            The import options.
+     */
+    public void importRelatedLearningObjects(Collection<? extends CourseCompetency> sourceCourseCompetencies, Map<Long, CompetencyWithTailRelationDTO> idToImportedCompetency,
+            Course courseToImportInto, CompetencyImportOptionsDTO importOptions) {
+        Set<CourseCompetency> importedCourseCompetencies = idToImportedCompetency.values().stream().map(CompetencyWithTailRelationDTO::competency).collect(Collectors.toSet());
+
+        Set<Exercise> importedExercises = new HashSet<>();
+        if (importOptions.importExercises()) {
+            importOrLoadExercises(sourceCourseCompetencies, idToImportedCompetency, courseToImportInto, importedExercises);
+        }
+        Map<String, Lecture> titleToImportedLectures = new HashMap<>();
+        Set<LectureUnit> importedLectureUnits = new HashSet<>();
+        if (importOptions.importLectures()) {
+            importOrLoadLectureUnits(sourceCourseCompetencies, idToImportedCompetency, courseToImportInto, titleToImportedLectures, importedLectureUnits);
+        }
+        Set<Lecture> importedLectures = new HashSet<>(titleToImportedLectures.values());
+
+        if (importOptions.referenceDate().isPresent()) {
+            setAllDates(importedExercises, importedLectures, importedLectureUnits, importedCourseCompetencies, importOptions.referenceDate().get(), importOptions.isReleaseDate());
+        }
+
+        courseCompetencyRepository.saveAll(importedCourseCompetencies);
+        exerciseRepository.saveAll(importedExercises);
+        lectureRepository.saveAll(importedLectures);
+    }
+
+    private void importOrLoadExercises(Collection<? extends CourseCompetency> sourceCourseCompetencies, Map<Long, CompetencyWithTailRelationDTO> idToImportedCompetency,
+            Course courseToImportInto, Set<Exercise> importedExercises) {
+        for (CourseCompetency sourceCourseCompetency : sourceCourseCompetencies) {
+            for (Exercise sourceExercise : sourceCourseCompetency.getExercises()) {
+                try {
+                    Exercise importedExercise = importOrLoadExercise(sourceExercise, courseToImportInto);
+
+                    importedExercises.add(importedExercise);
+
+                    importedExercise.getCompetencies().add(idToImportedCompetency.get(sourceCourseCompetency.getId()).competency());
+                    idToImportedCompetency.get(sourceCourseCompetency.getId()).competency().getExercises().add(importedExercise);
+                }
+                catch (Exception e) {
+                    // Catch exception, so we can still import the other learning objects
```

Comment:
is this comment necessary, i kind of feel like we can leave it out but dont have a strong opinion on it